cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -589,6 +582,13 @@ public void createApplicationVersion(ApplicationId id,
     if (latest != null) {
       List<Field<?>> fields = getApplicationPrimaryKeys(id.getNamespace(), id.getApplication(),
                                                         latest.getSpec().getAppVersion());
+      // Assign a creation time if it's null for the previous latest app version
+      // It is for the case that user has a pre-6.8 version, we mark it as past version (like created 1s ago)
+      // So it's sortable on creation time, especially when UI displays the version history for a pipeline
+      if (latest.getChange() == null || latest.getChange().getCreationTimeMillis() == 0) {
```

Comment:
it's not good to special case 0 like this just for the unit test. Change with time 0 won't actually be any real row that could ever have been written. Instead, the unit test should be writing a row just like the one before, where all the change information is null.

Reply:
@albertshau PTAL now, I've addressed the comments